### PR TITLE
👨🏻‍💻 enable controller register rules on each render

### LINKED ISSUE #1770: Controller rules reset
_| create a new issue, so we don't notify and annoy others._

I changed the previous code as much as i could to simulate my real problem. 
Can you take a look? https://codesandbox.io/s/controller-rules-npe18?file=/src/App.tsx

As you can see, required erros appear even though the fields are not required in that moment.
Thats why i thought on re-registering both hour fields whenever the switched are toggled.
Thanks.

_Originally posted by @pedroguia in https://github.com/react-hook-form/react-hook-form/issues/1749#issuecomment-638512677_

### PR DESCRIPTION
**Pull request by bluebill1049**
we had quite few issues/questions around change rules with controller due to cache inside the controller, i have been thinking over the nights about this issue and decided to take register each render as a tradeoff for exchange better DX, I did have documentation updated for V6 and the more I have to write about it make me think how unintuitive for a developer.

fix #1770 #1749

### CODE REVIEW

File: src/controller.tsx [hunk 1]
```diff
@@ -78,26 +76,30 @@ const Controller = <
   };
 
   const registerField = React.useCallback(() => {
-    if (!isNotFieldArray) {
-      removeFieldEventListener(fieldsRef.current[name] as Field, true);
-    }
+    if (fieldsRef.current[name]) {
+      register(name, rules);
```
Comment (bluebill1049) [inline src/controller.tsx:80]: if input already been registered, we just keep updating the rules.

**Comment by JeromeDeLeon** [comment]
Why close this one? What kind of rules are they talking about? is it about dynamic rules and since it cached inside rules, it didn't work as they expected? Am I right?

**Comment by bluebill1049** [comment]
Yes I tried, didn’t work well. Yes dynamic rules.

**Comment by bluebill1049** [comment]
i am going to take a look at it again.

**Pull request merged**